src/edgepi/dac/edgepi_dac.py

```diff
@@ -167,14 +195,40 @@ def compute_expected_voltage(self, analog_out: DACChannel) -> float:
         dac_gain = CalibConst.DAC_GAIN_FACTOR.value if self.__get_gain_state() else 1
         return self.dac_ops.code_to_voltage(analog_out.value, code, dac_gain)
 
-    def enable_dac_gain(self, enable: bool = None):
+    def __compute_ch_code_vals(self, enable: bool = None):
+        """
+        Compute the new code value to keep the current output voltage
+        Args:
+            enable(bool): False: multiply the current code value by 2
+                          True: divide the current code value by 2
+        Return:
+            code_vals (list): list of code values from ch1-ch8
+        """
+        code_vals = []
+        for ch in DACChannel:
+            code, _, _ = self.get_state(ch, True, False, False)
+            code_vals.append(int(code/CalibConst.DAC_GAIN_FACTOR.value) if enable else\
+                             code*CalibConst.DAC_GAIN_FACTOR.value if code<CalibConst.RANGE.value/2\
+                             else code)
```

Comment:
this input is quite long .. could it possibly be refactored into a function ?

Reply:
df7eb18d92d9516015533bb2808e871da631df3a